jobrunrpro/src/test/java/org/axonframework/extensions/jobrunrpro/deadline/JobRunrProDeadlineManagerBuilderTest.java

```diff
@@ -0,0 +1,117 @@
+
+import org.axonframework.common.AxonConfigurationException;
+import org.axonframework.common.transaction.TransactionManager;
+import org.axonframework.deadline.TestScopeDescriptor;
+import org.axonframework.messaging.ScopeAwareProvider;
+import org.axonframework.messaging.ScopeDescriptor;
+import org.axonframework.serialization.TestSerializer;
+import org.jobrunr.scheduling.JobScheduler;
+import org.jobrunr.storage.JobSearchRequest;
+import org.jobrunr.storage.Page;
+import org.jobrunr.storage.PageRequest;
+import org.jobrunr.storage.StorageProvider;
+import org.junit.jupiter.api.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class JobRunrProDeadlineManagerBuilderTest {
+
+    private static final String TEST_DEADLINE_NAME = "deadline-name";
+    private JobRunrProDeadlineManager.Builder builder;
+    private final JobScheduler jobScheduler = mock(JobScheduler.class);
+
+    private final StorageProvider storageProvider = mock(StorageProvider.class);
+    private final TransactionManager transactionManager = mock(TransactionManager.class);
+    private final ScopeAwareProvider scopeAwareProvider = mock(ScopeAwareProvider.class);
+
+    private final Page<?> page = mock(Page.class);
+
+    @BeforeEach
+    void newBuilder() {
+        builder = JobRunrProDeadlineManager.proBuilder();
+        doReturn(page).when(storageProvider).getJobs(any(JobSearchRequest.class), any(PageRequest.class));
+    }
+
+    @Test
+    void whenAllPropertiesAreSetCreatesManager() {
+        JobRunrProDeadlineManager manager =
+                builder.scopeAwareProvider(scopeAwareProvider)
+                       .transactionManager(transactionManager)
+                       .jobScheduler(jobScheduler)
+                       .storageProvider(storageProvider)
+                       .serializer(TestSerializer.JACKSON.getSerializer())
+                       .build();
+
+        assertNotNull(manager);
+    }
+
+    @Test
+    void validateNeedsAllPropertiesSet() {
+        builder.jobScheduler(jobScheduler)
+               .transactionManager(transactionManager);
+        assertThrows(AxonConfigurationException.class, () -> builder.build());
+    }
+
+    @Test
+    void whenSettingSchedulerWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.jobScheduler(null));
+    }
+
+    @Test
+    void whenSettingstorageProviderWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.storageProvider(null));
+    }
+
+    @Test
+    void whenSettingTransactionManagerWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.transactionManager(null));
+    }
+
+    @Test
+    void whenSettingScopeAwareProviderWithNullThrowError() {
+        assertThrows(AxonConfigurationException.class, () -> builder.scopeAwareProvider(null));
+    }
+
+    @Test
+    void cancelImplemented() {
```

Comment:
Nit: Being a bit of an ass here, but the `cancelImplemented` and the `cancelAllWithinScopeImplemented` aren't tests I'd put under the "builder" group. So, I would expect this class to be called `JobRunrProDeadlineManagerTest` instead.

Reply:
moved